packages/dhis2/src/http.js

```diff
@@ -0,0 +1,238 @@
+import { expandReferences, encode } from '@openfn/language-common/util';
+import * as util from './Utils';
+
+/**
+ * State object
+ * @typedef {Object} Dhis2State
+ * @property data - The response body (as JSON)
+ * @property response - The HTTP response from the Dhis2 server (excluding the body)
+ * @property references - An array of all previous data objects used in the Job
+ */
+
+/**
+ * Options object
+ * @typedef {Object} RequestOptions
+ * @property {object} query - An object of query parameters to be encoded into the URL
+ * @property {object} headers - An object of all request headers
+ * @property {string} [parseAs='json'] - The response format to parse (e.g., 'json', 'text', 'stream', or 'base64'. Defaults to `json`
+ * @property {string} [apiVersion=42] - The apiVersion of the request. Defaults to 42.
+ */
+
+/**
+ * Get data. Generic helper method for getting data of any kind from DHIS2.
```

Comment:
Not a "generic" helper. Maybe a HTTP helper.

How about this:

> HTTP helper method for fetching data from DHIS2. All paths are relative to baseUrl + version in your configuration

That pattern should be applied to all helpers